gobblin-hive-registration/src/main/java/gobblin/hive/HiveRegProps.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.hive;
+
+import java.util.List;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
+
+import gobblin.annotation.Alpha;
+import gobblin.configuration.State;
+import lombok.Getter;
+
+
+/**
+ * An extension to {@link State} for Hive registration.
+ *
+ * @author ziliu
+ */
+@Alpha
+public class HiveRegProps extends State {
+
+  public static final String HIVE_DB_ROOT_DIR = "hive.db.root.dir";
+  public static final String HIVE_OWNER = "hive.owner";
+  public static final String HIVE_REGISTER_THREADS = "hive.register.threads";
+  public static final int DEFAULT_HIVE_REGISTER_THREADS = 20;
+  public static final String HIVE_TABLE_PROPS = "hive.table.props";
+
+  private static final Splitter SPLITTER = Splitter.on('=').trimResults().omitEmptyStrings();
+
+  @Getter
+  private final State tableProps;
+
+  public HiveRegProps(State props) {
```

Comment:
Because if we run it on Azkaban, then the program starts with a single property object from Azkaban. Someone has to separate tableProps from the property object somewhere, if not here then somewhere else. I don't see a reason why it should be somewhere else.